### PR TITLE
unicode-emoji: 12.1 -> 14.0; cldr-common: init at 40.0; ibus: update for Emoji 14

### PR DESCRIPTION
**Pull request by DeeUnderscore**
###### Description of changes
The goal of this PR is to get ibus to support Emoji 14 in its emoji picker. To this end, this PR:
* bumps `unicode-emoji` to 14.0
* adds `cldr-common`. The ibus package currently uses `cldr-emoji-annotations`, which is no longer maintained (per [the README](https://github.com/fujiwarat/cldr-emoji-annotation/blob/master/README)). As the README suggests, I switched over to upstream CLDR tarballs.

I built the ibus with the changes and tested out its emoji picker, but I didn't rebuild all the packages (~50) dependent on ibus. 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ttuegel — COMMENTED** [review]
Can we remove `cldr-emoji-annotation`?

**Comment by DeeUnderscore** [comment]
`fcitx5` currently also uses `cldr-emoji-annotations`. It could probably be replaced with `cldr-common` there, too, but I'm not as familiar with fcitx, so I haven't looked into it more closely, yet.

File: pkgs/data/misc/cldr-common/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ stdenvNoCC
+, lib
+, fetchzip }:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "cldr-common";
+  version = "40.0";
+
+  src = fetchzip {
+    url = "http://unicode.org/Public/cldr/40/cldr-common-${version}.zip";
```
Comment (SuperSandro2000) [inline pkgs/data/misc/cldr-common/default.nix:10]:
```suggestion
    url = "https://unicode.org/Public/cldr/40/cldr-common-${version}.zip";
```

File: pkgs/data/misc/cldr-common/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{ stdenvNoCC
+, lib
+, fetchzip }:
+
+stdenvNoCC.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/data/misc/cldr-common/default.nix:5]: If we are just copying files we can also use fetchzip entirely.

File: pkgs/data/misc/cldr-common/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{ stdenvNoCC
+, lib
+, fetchzip }:
```
Comment (SuperSandro2000) [inline pkgs/data/misc/cldr-common/default.nix:3]:
```suggestion
, fetchzip
}:
```

**Comment by DeeUnderscore** [comment]
I've looked into using this for fcitx5, and unlike ibus, fcitx5 uses the annotation files directly. Since both ibus and fcitx only use annotation data from CLDR, I can probably avoid adding >200 MB to the closure by only using the annotation files from the CLDR zip, so I'm gonna look into doing that instead.

**Review by ttuegel — APPROVED** [review]

**Comment by ttuegel** [comment]
Thanks!

**Pull request merged**